hazelcast/src/main/java/com/hazelcast/internal/networking/nio/NioEventLoopGroup.java

```diff
@@ -271,31 +250,64 @@ private void shutdown(NioThread[] threads) {
     }
 
     @Override
-    public ChannelWriter newSocketWriter(ChannelConnection connection) {
+    public void register(final Channel channel) {
+        NioChannel nioChannel = checkInstanceOf(NioChannel.class, channel);
+
+        try {
+            nioChannel.socketChannel().configureBlocking(false);
+        } catch (IOException e) {
+            throw rethrow(e);
+        }
+
+        final NioChannelReader reader = newChannelReader(nioChannel);
+        final NioChannelWriter writer = newChannelWriter(nioChannel);
+
+        nioChannel.setReader(reader);
+        nioChannel.setWriter(writer);
+
+        ioBalancer.channelAdded(reader, writer);
+
+        String metricsId = channel.getLocalSocketAddress() + "->" + channel.getRemoteSocketAddress();
+        metricsRegistry.scanAndRegister(writer, "tcp.connection[" + metricsId + "].out");
```

Comment:
In the future we can remove 'tcp.connection' into a prefix variable. Currently both client/member connections are tcp.connections.